### DISCUSSION TITLE
Home Assistant not automatically detecting MQTT broker

### DISCUSSION THREAD

**Discussion by focher** [Resinchem/Tailwind2MQTT#7] [Q&A] [answered]
I currently run Home Assistant Container under Docker on a QNAP NAS. It's on my "internal" home network. I also run a separate segregated WiFi network called "IoT" where pretty much all my home automation devices are connected.

I have `ha_discovery = true` but Home Assistant is not automatically detecting anything. 

I have firewall rules in place that should allow all traffic from "Internal" to "IoT", but only existing connections are open from "IoT" to "Internal".

Any ideas why Home Assistant is not seeing the MQTT broker?

**Comment by Resinchem** [comment, marked as answer]
I'm guessing you are using MQTT for other things in HA and have the broker defined in HA?  Are you using the MQTT add-on for Home Assistant or a standalone broker?  In my case, I'm running the full supervised HA in a Proxmox VM... using the Mosquito MQTT add-on.  Can you use something like MQTT Explorer connected to your broker and see if the Tailwind messages are being received by the broker?  This will let us know that at least the bridge is talking to the broker and that the real issue is the connection between HA and the broker.  Note that since commands are not retained in MQTT, you will only see those appear in Explorer if you reboot the bridge after connecting to MQTT explorer... but the states should show regardless.

**Reply by focher** [reply]
No, this is the first MQTT integration - both in Home Assistant and me personally. Maybe that's why I'm a bit confused.

It appears that, because I'm using Home Assistant Container and don't have Supervisor, I cannot install Add-ons. Do you happen to know any alternative?

I'm tempted to take one of my Raspberry Pis and deploy the full Home Assistant Operating System.

**Reply by Resinchem** [reply]
Ah... makes sense.  I'm guessing you have an external broker in it's own docker container then?  If not, then see here: [Eclipse Mosquitto Docker image](https://hub.docker.com/_/eclipse-mosquitto).  You absolutely do not have to use the Home Assistant add-on, but you do need to define the connection to your external MQTT broker so that Home Assistant can talk to it.  If you can't use the provided integration, worst case, you can define your MQTT broker in your configuration.yaml  See [Home Assistant MQTT](https://www.home-assistant.io/integrations/mqtt/) for info.

I will add that I was a bit lost with MQTT at first (topics, payloads, publish, subscribe, retained, etc.), but now that I've gotten use to it, it is by far my first choice for integrations if it is available.  It is lightweight, fast, reliable and you have full control.  And if you don't have it, I'd highly recommend [MQTT-Explorer](http://mqtt-explorer.com/).  It's available on all platforms and is an invaluable tool when troubleshooting MQTT as it allows you to see anything moving in or out of your broker.  You can also issue MQTT commands to devices.  Again, I appreciate you taking the time, so don't hesitate to reach out if I can be of any additional help on the MQTT front!

**Reply by focher** [reply]
Thanks so much for your patience but I finally got there. Here's what I did:

1. Deployed Mosquitto MQTT Broker on Docker (Internal network) with user credentials of "mqtt".
2. Configured the Tailwind2MQTT with the user credentials (IoT network).
3. Opened port 1883 from Tailwind2MQTT.

Magically, Home Assistant immediately saw everything and I was able to add a card to one of my Lovelace dashboards.

MQTT Explorer is helpful, for sure, because it confirmed the MQTT Broker was up and running.

Thanks again!

I'll definitely now submit any feedback or issues I experience.

**Reply by Resinchem** [reply]
Great!  Glad I could help out.  I think that I could probably clarify some of the info in the wiki based on your experience.  I wrote it strictly from my own perspective and really didn't consider different types of Home Assistant installs, those that might not have MQTT, etc.  And once Scott and the guys from Tailwind implement security, I'd like to turn this into a downloadable .bin file so that people don't have to go through the compiling, Arduino libraries, etc. and just upload it to the D1 Mini.  Let me know if you see any issues or have additional thoughts.  I really like the fact that I can now include Tailwind in my HA automations.
